lib/assessment.rb

```diff
@@ -5,35 +5,48 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
-##
-# your code here
-##
+#
+class Dog
+  attr_reader :name, :breed, :age
+
+  def initialize(name, breed, age)
+    @name, @breed, @age = name, breed, age
+  end
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Answer.dog = Dog.new('Lassie', 'collie', 5)
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
-##
-# your code here
-##
+#
+class Person
+  attr_writer :name, :location
```

Comment:
Missed this earlier, but you should be using `attr_accessor` here, not `attr_writer`. `attr_writer` is almost never used, because it creates a 'setter' method _but not a 'getter one_ - in effect, it gives write access without giving read access.